sdk/monitor/monitor-opentelemetry/src/shared/config.ts

```diff
@@ -91,7 +91,9 @@ export class InternalConfig implements AzureMonitorOpenTelemetryOptions {
         options.instrumentationOptions,
       );
       this.resource = Object.assign(this.resource, options.resource);
-      this.samplingRatio = options.samplingRatio || this.samplingRatio;
+      if (typeof options.samplingRatio === "number") {
```

Comment:
```suggestion
      if (typeof options.samplingRatio === "number" && Number.isFinite(options.samplingRatio)) {
```

Reply:
Thanks! I've included this check in the `sampler.ts` file since my thinking is that it's possible the user could be doing some arithmetic calculation to determine a sampling ratio, if that calculation fails and produces `NaN`, `Infinity` or `-Infinity` it'd be more helpful to provide them with the specific "Invalid sampling rate" error than to just go ahead and set sampling to 1 anyway.

Reply:
The config value needs to be a number between 0 and 1, so better to check that explicitly here, @JacksonWeber we should log a warning if config is incorrect in both JSON or JS configuration provided by the user, looks like sampler is actually throwing an error when incorrect sampling rate is provided, please remove that, this could cause some unhandled exception and affect customers applications.

Reply:
@hectorhdzg I updated the logic in the `ApplicationInsightsSampler` to log a warning instead of throwing an uncaught exception. I also unified most of the "invalid sampling ratio" logic there. This check was just intended to avoid the scenario where a customer passes in 0 which was previously treated as falsy. 
Keeps the majority of our samplingRatio logic in one place, and ensures that no matter how configuration might be modified in future, that the `ApplicationInsightsSampler` class will always catch any invalid sampling ratios.